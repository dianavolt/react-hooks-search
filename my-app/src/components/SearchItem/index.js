import React from 'react'
import PropTypes from 'prop-types'

import './styles.scss'

const SearchItems = ({ id, label }) => {
    return (
        <div className="search-item">
            <a href={id} target="_blank">
                {label}
            </a>
            <p>{id}</p>
        </div>
    )
}

SearchItems.propTypes = {
    id: PropTypes.string.isRequired,
    label:PropTypes.string.isRequired
}

export default SearchItems
