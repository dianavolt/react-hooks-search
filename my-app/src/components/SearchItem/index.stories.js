import React from 'react';

import SearchItem from '.';

export default {
  title: 'Components/SearchItem',
  component: SearchItem,
}

const Template = (args) => <SearchItem {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  id: 'https://en.wikipedia.org/wiki/Google',
  label: 'Test data for search'
}