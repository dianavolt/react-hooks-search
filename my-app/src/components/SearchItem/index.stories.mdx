import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
import SearchItem from '.';

<Meta title="Components/SearchItem" component={SearchItem} />

# Search SearchItem

With `MDX` we can define a story for `SearchItem` right in the middle of our
Markdown documentation.

<!--- This is your Story template function, shown here in React -->

export const Template = (args) => <SearchItem {...args} />

<Canvas>
  <Story name="Default" args={{
      label: 'Default story',
      id: 'https://id.com/1'
    }}>
    {Template.bind({})}
  </Story>
</Canvas>
