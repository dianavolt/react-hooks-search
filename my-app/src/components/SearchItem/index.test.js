import React from 'react'
import { shallow } from 'enzyme'

import SearchItem from '.'

const render = (props) => shallow(<SearchItem {...props}/>)

describe('SearchItem component', () => {
    let test
    let props

    describe('withuot props', () => {
        beforeEach(() => {
            test = render(props)
        })
        
        it('should match snapshot', () => {
            expect(test).toMatchSnapshot()
        })
    })

    describe('with props', () => {
        beforeEach(() => {
            props = {
                id: 'testID123',
                label: 'searchTerm'
            }

            test = render(props)
        })

        it('should match snapshot', () => {
            expect(test).toMatchSnapshot()
        })
    })
})
