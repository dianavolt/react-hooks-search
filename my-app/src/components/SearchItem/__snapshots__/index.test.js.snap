// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem component with props should match snapshot 1`] = `
<div
  className="search-item"
>
  <a
    href="testID123"
    target="_blank"
  >
    searchTerm
  </a>
  <p>
    testID123
  </p>
</div>
`;

exports[`SearchItem component withuot props should match snapshot 1`] = `
<div
  className="search-item"
>
  <a
    target="_blank"
  />
  <p />
</div>
`;
